components/org.wso2.identity.outbound.adapter.websubhub/src/main/java/org/wso2/identity/outbound/adapter/websubhub/internal/WebSubHubAdapterServiceComponent.java

```diff
@@ -75,6 +79,21 @@ protected void deactivate(ComponentContext context) {
             log.debug("Successfully de-activated the WebSub Hub adapter service.");
         }
     }
-}
 
+    @Reference(
+            name = "registry.service",
+            service = RegistryService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetRegistryService"
+    )
+    protected void setRegistryService(RegistryService registryService) {
```

Comment:
Can we make sure that RegistryService transitively depends on the CarbonCore to verify that the CarbonCore will always be initialized before this component? If not, lets change this to a framework level service.

Reply:
Fixed